toolchain/sem_ir/file.cpp

```diff
@@ -13,9 +13,26 @@
 #include "toolchain/sem_ir/ids.h"
 #include "toolchain/sem_ir/inst.h"
 #include "toolchain/sem_ir/inst_kind.h"
+#include "toolchain/sem_ir/typed_insts.h"
 
 namespace Carbon::SemIR {
 
+auto Function::GetParamIdFromParamRefId(const File& sem_ir, InstId param_ref_id)
+    -> InstId {
+  auto ref = sem_ir.insts().Get(param_ref_id);
+
+  if (auto addr_pattern = ref.TryAs<SemIR::AddrPattern>()) {
+    param_ref_id = addr_pattern->inner_id;
+    ref = sem_ir.insts().Get(param_ref_id);
+  }
+
+  if (auto bind_name = ref.TryAs<SemIR::BindName>()) {
+    param_ref_id = bind_name->value_id;
+  }
+
+  return param_ref_id;
```

Comment:
```suggestion
  CARBON_CHECK(ref.Is<SemIR::Param>());
  return param_ref_id;
```

Note, I'm suggesting this mainly because I was trying to figure out whether a specific type of instruction should be returned, and a CHECK would've helped me be confident. The comment change in the header may be sufficient; this might be undesirable since it kind of wastes an insts() query.

Reply:
We now call `As<SemIR::Param>` which implicitly performs this check.